pkg/eventgenerator/types.go

```diff
@@ -0,0 +1,43 @@
+package eventgenerator
+
+import (
+	"fmt"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/eventgenerator/dns"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/eventgenerator/http"
+)
+
+// Generator defines the interface for event generators
+type Generator interface {
+	Generate(target, countStr, intervalStr string) (string, string, string, error)
+	Cleanup() (string, error)
+}
+
+// NewGenerator creates a new event generator based on the type
+func NewGenerator(eventType string, log logger.Logger) (Generator, error) {
+	config, err := getKubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Kubernetes config: %v", err)
+	}
+
+	switch eventType {
+	case "dns":
+		return dns.NewGenerator(config, log)
+	case "http":
+		return http.NewGenerator(config, log)
+	default:
+		return nil, fmt.Errorf("unknown generator type: %s", eventType)
+	}
+}
+
+func getKubeConfig() (*rest.Config, error) {
+	config, err := rest.InClusterConfig()
```

Comment:
got this error when using ConfigFlags 
```
go run --exec "sudo -E" ./cmd/ig run trace_dns --filter name==example.com. --eventgen-enable=true --eventgen-type=dns --eventgen-count=5 --eventgen-interval=5 --eventgen-target=example.com
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME          SRC                                     DST                                     COMM                   PID QR QTYPE           NAME                          RCODE    ADDRESSES      
WARN[0011] Replacement map "gadget_sockets" incompatible: ValueSize: 8272 changed to 64: map spec is incompatible with existing map 
WARN[0011] Is trace_dns compatible with ig 0.0.0?       
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf0 pc=0x2e1c04e]

goroutine 1 [running]:
k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).ToRawKubeConfigLoader(0xc00545f530?)
	/home/h4logen/go/pkg/mod/k8s.io/cli-runtime@v0.31.1/pkg/genericclioptions/config_flags.go:152 +0xe
k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).ToRESTConfig(0x0)
	/home/h4logen/go/pkg/mod/k8s.io/cli-runtime@v0.31.1/pkg/genericclioptions/config_flags.go:138 +0x18
github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil.NewClientsetFromConfigFlags(0x8?)
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/k8sutil/client.go:85 +0x13
github.com/inspektor-gadget/inspektor-gadget/pkg/eventgenerator/dns.NewGenerator(0x0?, {0x4b50008, 0x6776ec0})
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/eventgenerator/dns/generator.go:25 +0x25
github.com/inspektor-gadget/inspektor-gadget/pkg/eventgenerator.NewGenerator({0x7ffc4689d0f1?, 0x4716403?}, 0x10?, {0x4b50008?, 0x6776ec0?})
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/eventgenerator/types.go:21 +0x45
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/eventgen.(*eventGenOperatorInstance).Start(0xc004907050, {0x4b4e430, 0xc0000dd040})
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/operators/eventgen/operator.go:88 +0x15c
github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context.(*GadgetContext).start(0xc0000dd040, {0xc0049163c0, 0x8, 0xc0089086a8?})
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/gadget-context/run.go:109 +0x2a8
github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context.(*GadgetContext).Run(0xc0000dd040, 0xc00061c7e0?)
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/gadget-context/run.go:156 +0xc8
github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local.(*Runtime).RunGadget(0x4072ba0?, {0x4b80ae0?, 0xc0000dd040?}, 0xd?, 0xc0086ebb28?)
	/home/h4logen/new-ig-arch/inspektor-gadget/pkg/runtime/local/oci.go:35 +0x27
github.com/inspektor-gadget/inspektor-gadget/cmd/common.NewRunCommand.func2(0xc000626f08, {0xc0007b2e80?, 0x8?, 0x8?})
	/home/h4logen/new-ig-arch/inspektor-gadget/cmd/common/oci.go:354 +0x13e2
github.com/spf13/cobra.(*Command).execute(0xc000626f08, {0xc0007b2e80, 0x8, 0x8})
	/home/h4logen/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc00049c908)
	/home/h4logen/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/h4logen/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	/home/h4logen/new-ig-arch/inspektor-gadget/cmd/ig/main.go:109 +0x4ef
exit status 2
```
because configFlags is nil when we try to use it. I guess this is because `utils.KubernetesConfigFlags` is only available when using `kubectl-gadget`.

Reply:
Oh! If we want talk to API server using ig perhaps we also need to expose the Kubernetes flags. So maybe keep it as it for now but add comment saying why we couldn't use k8sutils. 

Also, please add a comment to https://github.com/inspektor-gadget/inspektor-gadget/issues/3603 that we should handle K8s flags as part of improving how ig talks to API server?